diffsmhm/diff_stats/cuda/wprp.py

```diff
@@ -478,13 +473,8 @@ def wprp_mpi_kernel_cuda(
     """
     # check if cupy is available
     # this is here bc github CI doesn't work with cupy currently
-    try:
-        _ = cp.array([1])
-        can_cupy = True
-        xp = cp
-    except RuntimeError:
-        can_cupy = False
-        xp = np
+    xp = cp.get_array_module(x1)
+    can_cupy = xp == cp
```

Comment:
you may want `xp is cp` here. This tests object identity as opposed to equality which is mediated through the `__eq__` magic methods and such.